sdk/search/Azure.Search.Documents/tests/Utilities/SearchResources.cs

```diff
@@ -442,6 +547,90 @@ private async Task<SearchResources> CreateSearchServiceIndexAndDocumentsAsync()
             return this;
         }
 
+        /// <summary>
+        /// Creates a new storage account.
+        /// </summary>
+        /// <returns>The current <see cref="SearchResources"/>.</returns>
+        private async Task<SearchResources> CreateStorageAccountAsync()
+        {
+            if (TestFixture.Mode != RecordedTestMode.Playback)
+            {
+                using CancellationTokenSource cts = new CancellationTokenSource(Settings.Timeout);
+
+                StorageManagementClient client = GetStorageManagementClient();
+
+                // Use same resource name but without dashes that are invalid for storage account names.
+                StorageAccountName = Settings.ResourcePrefix.Replace("-", string.Empty) + Settings.Random.GetName(8);
+                StorageAccountCreateParameters parameters = new StorageAccountCreateParameters
+                {
+                    AccessTier = AccessTier.Hot,
+                    Kind = Microsoft.Azure.Management.Storage.Models.Kind.BlobStorage,
+                    Location = Settings.Location,
+                    Sku = new Microsoft.Azure.Management.Storage.Models.Sku
+                    {
+                        Name = Microsoft.Azure.Management.Storage.Models.SkuName.StandardLRS,
+                    },
+                };
+
+                StorageAccount account = await client.StorageAccounts.CreateAsync(
+                    Settings.ResourceGroup,
+                    StorageAccountName,
+                    parameters,
+                    cts.Token);
+
+                StorageAccountListKeysResult keys = await client.StorageAccounts.ListKeysAsync(
+                    Settings.ResourceGroup,
+                    account.Name,
+                    cts.Token);
+                StorageAccountKey key = keys.Keys[0];
+
+                StorageAccountConnectionString = $"DefaultEndpointsProtocol=https;AccountName={account.Name};AccountKey={key.Value};EndpointSuffix=core.windows.net";
+            }
+
+            return this;
+        }
+
+        /// <summary>
+        /// Upload <see cref="TestDocuments"/> to a new blob storage container named "hotels".
+        /// </summary>
+        /// <returns>The current <see cref="SearchResources"/>.</returns>
+        private async Task<SearchResources> CreateHotelsContainerAsync()
+        {
+            if (TestFixture.Mode != RecordedTestMode.Playback)
+            {
+                using CancellationTokenSource cts = new CancellationTokenSource(Settings.Timeout);
+
+                BlobContainerClient client = new BlobContainerClient(StorageAccountConnectionString, "hotels");
+                await client.CreateAsync(cancellationToken: cts.Token);
+
+                Hotel[] hotels = TestDocuments;
+                List<Task> tasks = new List<Task>(hotels.Length);
+
+                foreach (Hotel hotel in hotels)
+                {
+                    Task task = Task.Run(async () =>
+                    {
```

Comment:
The body looks like it should be fine directly in the `foreach`.

Reply:
I don't want to execute them now in sequence. That's why create and add a task to the list to execute in parallel below in `await Task.WhenAll`. Faster this way, or is there some batching you're doing in the SDK that will do this in bulk already?

Reply:
I see - makes sense.